domain/status/state/storage_test.go

```diff
@@ -0,0 +1,537 @@
+		MinJujuVersion: semversion.MustParse("4.0.0"),
+		Assumes:        []byte("null"),
+	}, nil
+}
+
+func (s *storageSuite) insertCharmWithStorage(c *tc.C, stor ...charmStorageArg) corecharm.ID {
+	uuid := charmtesting.GenCharmID(c)
+
+	err := s.TxnRunner().StdTxn(c.Context(), func(ctx context.Context, tx *sql.Tx) error {
+		var err error
+		if _, err = insertCharmMetadata(c, tx, uuid); err != nil {
+			return errors.Capture(err)
+		}
+
+		for _, arg := range stor {
+			_, err = tx.ExecContext(ctx, `
+INSERT INTO charm_storage (
+    charm_uuid,
+    name,
+    storage_kind_id,
+    read_only,
+    count_min,
+    count_max,
+    location
+) VALUES
+    (?, ?, ?, ?, ?, ?, ?);`,
+				uuid, arg.name, arg.kind, arg.readOnly, arg.min, arg.max, arg.location)
+			if err != nil {
+				return errors.Capture(err)
+			}
+		}
+		return nil
+	})
+	c.Assert(err, tc.ErrorIsNil)
+	return uuid
+}
+
+func (s *storageSuite) createStorageInstance(c *tc.C, storageName, charmUUID corecharm.ID) corestorage.UUID {
+	ctx := c.Context()
+	storageUUID := storagetesting.GenStorageUUID(c)
+	err := s.TxnRunner().StdTxn(ctx, func(ctx context.Context, tx *sql.Tx) error {
+		_, err := tx.ExecContext(ctx, `
+INSERT INTO storage_instance (
+    uuid, charm_uuid, storage_name, storage_id,
+    life_id, requested_size_mib, storage_type
+) VALUES (?, ?, ?, ?, ?, ?, ?)`,
+			storageUUID, charmUUID, storageName,
+			fmt.Sprintf("%s/%d", storageName, s.storageInstCount),
+			life.Alive, 100, "pool")
+		return err
+	})
+	c.Assert(err, tc.ErrorIsNil)
+
+	s.storageInstCount = s.storageInstCount + 1
+	return storageUUID
+}
+
+func (s *storageSuite) createFilesystemInstance(c *tc.C, filesystemUUID corestorage.FilesystemUUID) {
+	charmUUID := s.insertCharmWithStorage(c, filesystemStorage)
+	storageUUID := s.createStorageInstance(c, "pgdata", charmUUID)
+
+	err := s.TxnRunner().StdTxn(c.Context(), func(ctx context.Context, tx *sql.Tx) error {
+		_, err := tx.ExecContext(ctx, `
+INSERT INTO storage_instance_filesystem(storage_filesystem_uuid, storage_instance_uuid)
+VALUES (?, ?)`, filesystemUUID, storageUUID)
+		return err
+	})
+	c.Assert(err, tc.ErrorIsNil)
+}
+
+func (s *storageSuite) assertFilesystemStatus(c *tc.C, filesystemUUID corestorage.FilesystemUUID, expected status.StatusInfo[status.StorageFilesystemStatusType]) {
+	ctx := c.Context()
+
+	var got status.StatusInfo[status.StorageFilesystemStatusType]
+	err := s.TxnRunner().StdTxn(ctx, func(ctx context.Context, tx *sql.Tx) error {
+		err := tx.QueryRowContext(ctx, `
+SELECT status_id, message, updated_at FROM storage_filesystem_status
+WHERE filesystem_uuid = ?`, filesystemUUID).Scan(
+			&got.Status, &got.Message, &got.Since,
+		)
+		if err != nil {
+			return err
+		}
+		return nil
+	})
+	c.Assert(err, tc.ErrorIsNil)
+	c.Check(got.Status, tc.DeepEquals, expected.Status)
+	c.Check(got.Message, tc.DeepEquals, expected.Message)
+	c.Check(got.Since, tc.NotNil)
+	c.Assert(*got.Since, tc.Before, time.Now().UTC())
```

Comment:
This line tests nothing. It is the caller to the state layer and these tests that get to dictate the Since value. This is a service level check and concern.

Reply:
Was meant to change this but it got missed. Thanks for the pick up.